File: sclib.js/EventManager.js
export class EventManager {
    #callbacks;

    constructor() {
        this.#callbacks = {};
    }

    add(keyword, callback) {
        if (!this.#callbacks[keyword]) {
            this.#callbacks[keyword] = [];
        }
        
        this.#callbacks[keyword].push({ callback });
    }

    call(keyword, ...args) {
        if (!this.#callbacks[keyword]) {
            return;
        }

        const callbacksToRemove = [];

        this.#callbacks[keyword].forEach(async (item, index) => {
            const rac = item.callback(...args);

            if (rac === true) {
                callbacksToRemove.push(index);
            }
        });

        callbacksToRemove.reverse().forEach(index => {
            this.#callbacks[keyword].splice(index, 1);
        });

        if (this.#callbacks[keyword].length === 0) {
            delete this.#callbacks[keyword];
        }
    }
} 

File: pysclib/event_manager.py
import types

class event_manager:
    def __init__(self):
        self._callbacks = {}
        self.is_debug = True

    def add(self, keyword, callback):
        if keyword not in self._callbacks:
            self._callbacks[keyword] = []
        
        self._callbacks[keyword].append({"callback": callback})

    def call(self, keyword, *args):
        if keyword not in self._callbacks:
            return

        callbacks_to_remove = []

        for index, item in enumerate(self._callbacks[keyword]):            
            if isinstance(item['callback'], types.FunctionType):
                if self.is_debug is True:
                    print(item["callback"])
                
                rac = item["callback"](*args)

                if rac is True:
                    callbacks_to_remove.append(index)

        for index in reversed(callbacks_to_remove):
            del self._callbacks[keyword][index]

        if len(self._callbacks[keyword]) == 0:
            del self._callbacks[keyword]